substrate/frame/migrations/src/benchmarking.rs

```diff
@@ -204,6 +207,34 @@ mod benches {
 		);
 	}
 
+	#[benchmark(skip_meta, pov_mode = Measured)]
+	fn reset_pallet_migration(n: Linear<0, 2048>) -> Result<(), BenchmarkError> {
+		let prefix: [u8; 16] = twox_128(b"__ResetPalletBenchmarkPrefix__");
+
+		for i in 0..n {
+			// we need to avoid allocations here
+			let mut iter = prefix.into_iter().chain(i.to_le_bytes());
+			let key: [u8; 20] = array::from_fn(|_| iter.next().unwrap());
+			// 32 byte will trigger the worst case where the value is
+			// no longer stored inline
+			storage::set(&key, &[0u8; 32]);
+		}
+
+		let result;
+
+		#[block]
+		{
+			result = storage::clear_prefix(&prefix, None);
+		}
+
+		match result {
+			KillStorageResult::AllRemoved(i) if i == n => (),
+			_ => Err("Unexpected number of keys were removed")?,
+		}
```

Comment:
It always comes back with `AllRemoved(0)`. I assume that is because the keys are still in the overlay. Is there a way to flush the overlay after writing to it? Or does it even make sense in the benchmark?